### PR TITLE
Vb 2080 category grouping update logic to use grouping similar to housing levels

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

update logic to use grouping similar to housing levels instead of inclusive / exclusive logic

## What is the intent behind these changes?

update logic to use grouping similar to housing levels

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt [hunk 1]
```diff
@@ -40,9 +40,6 @@ data class UpdateSessionTemplateDto(
   @Schema(description = "biWeekly time table", example = "true", required = true)
   val biWeekly: Boolean? = null,
 
-  @Schema(description = "list of included prisoner categories", required = false)
-  val includedPrisonerCategories: List<String>? = null,
-
-  @Schema(description = "list of excluded prisoner categories", required = false)
-  val excludedPrisonerCategories: List<String>? = null,
+  @Schema(description = "list of group references for allowed prisoner category groups", required = false)
+  val categoryGroupReferences: List<String>? = listOf(),
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt:44]: Should this be PrisonerCategory

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt [hunk 2]
```diff
@@ -56,9 +56,6 @@ data class CreateSessionTemplateDto(
   @Schema(description = "biWeekly time table", example = "true", required = true)
   val biWeekly: Boolean,
 
-  @Schema(description = "list of included prisoner categories", required = false)
-  val includedPrisonerCategories: List<String> = listOf(),
-
-  @Schema(description = "list of excluded prisoner categories", required = false)
-  val excludedPrisonerCategories: List<String> = listOf(),
+  @Schema(description = "list of group references for allowed prisoner category groups", required = false)
+  val categoryGroupReferences: List<String>? = listOf(),
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt:60]: Should this be PrisonerCategory

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/category/PrisonerCategory.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.model.entity.session.category
+
+/*
+TODO - this needs to be reconsidered based on tests and further analysis and might need to be replaced with a String.
+ */
+enum class PrisonerCategory(
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/category/PrisonerCategory.kt:6]: We have generally call these PrisonerCategoryType but your call

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt:44]: Based this on the locationGroupReferences in the same class. I thought it takes the group references as a list.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/CreateSessionTemplateDto.kt:60]: same as above.

**Review by AledWEvans — DISMISSED** [review]
LGTOM would recommend changing string types to enums in admin api

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/category/PrisonerCategory.kt (hunk 3 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/category/PrisonerCategory.kt:6]: will update to PrisonerCategoryType

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt (hunk 1 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt:44]: yes I was being stupid

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**